### PR TITLE
[Clean code] Equal(true/false) => BeTrue/BeFalse

### PR DESCRIPTION
**Pull request by nunnatsa**
Fix all cases with gomaga's Equal assertion to a boolean values.

`Equal(true)` ahould be `BeTrue()`
`Equal(false)` ahould be `BeFalse()`

/sig code-quality

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

```release-note
None
```

### CODE REVIEW

File: tests/infra_test.go [hunk 1]
```diff
@@ -1268,9 +1268,8 @@ var _ = Describe("[Serial][sig-compute]Infrastructure", func() {
 				for key := range node.Labels {
 					if strings.Contains(key, v1.CPUModelLabel) {
 						model := strings.TrimPrefix(key, v1.CPUModelLabel)
-						for obsoleteModel := range nodelabellerutil.DefaultObsoleteCPUModels {
-							Expect(model == obsoleteModel).To(Equal(false), "Node can't contain label with cpu model, which is in default obsolete filter")
-						}
+						Expect(nodelabellerutil.DefaultObsoleteCPUModels).ToNot(HaveKey(model),
+							"Node can't contain label with cpu model, which is in default obsolete filter")
```
Comment (orelmisan) [inline tests/infra_test.go:1272]: This change is different from the others, and is not documented in the commit message / PR description.
Do you mind doing it in a different PR?

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @nunnatsa.

File: tests/infra_test.go (hunk 1 above)
Comment (nunnatsa) [reply tests/infra_test.go:1272]: It's not different. I find it by searching `Equal(false)`, and the purpose is to get rid of this pattern. But just replacing it with `BeFalse()` in this case was wrong too, because this code could be simplified.

Comment (orelmisan) [reply tests/infra_test.go:1272]: I share your opinion that this change simplifies and improves the code, but it is not a simple replacement as depicted in the PR title and description.
IMO this change should get (at least) a commit of its own with a commit message which describes it:
> just replacing it with BeFalse() in this case was wrong too, because this code could be simplified.

Comment (nunnatsa) [reply tests/infra_test.go:1272]: @orelmisan - thanks. I guess it's a semantic discussion... Anyway, I split the original commit to two commits, as you offered.

**Review by orelmisan — APPROVED** [review]
Thank you for the change @nunnatsa

**Comment by rmohr** [comment]
@nunnatsa will your linter catch these in the future right away?
